keras/random/random_test.py

```diff
@@ -257,3 +257,27 @@ def test_gamma(self, seed, shape, alpha, dtype):
         self.assertEqual(ops.shape(values), shape)
         self.assertEqual(backend.standardize_dtype(values.dtype), dtype)
         self.assertGreater(np.min(ops.convert_to_numpy(values)), 0.0)
+
+    @parameterized.parameters(
+        {"seed": 10, "shape": (5, 2), "counts": 1., "probs": 0.5, "dtype": "float16"},
+        {"seed": 10, "shape": (2,), "counts": 1, "probs": 0.5, "dtype": "float32"},
+        {"seed": 10, "shape": (2, 3), "counts": np.array([[1, 2, 3], [4, 5, 6]]),
+         "probs": np.array([[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]]), "dtype": "float32"},
+    )
+    def test_binomial(self, seed, shape, counts, probs, dtype):
+        values = random.binomial(shape, counts=counts,
+                                 probs=probs, seed=seed, dtype=dtype)
+        self.assertEqual(ops.shape(values), shape)
+        self.assertEqual(backend.standardize_dtype(values.dtype), dtype)
```

Comment:
This makes no correctness check. Can we add one (even if statistical in nature)

Reply:
Yes we definitely can. I'll do that. Sorry at the time I didn't think that testing for the correctness of values generated by an underlying framework would be as important but now that I think about it, it makes loads of sense to do as it'd catch any potential change in the underlying framework or maybe an error in calling the function etc.

Reply:
Maybe we could do a certain number of draws and check the statistical properties of the outputs -- and make the test seeded to avoid flakiness

Reply:
Thank you. That makes a lot of sense. Additionally, I'll also do my own research as well to come up with a foolproof way of ensuring the correctness of results for both distributions.

Reply:
Hi,
I've come up with two tests to ensure the correctness of binomial values as follows. (I haven't commited/pushed these changes yet)

**Test 1:** Makes sure that the number of times an event occurs doesn't exceed the total possible count for that event specified by the user since the probability value for each event is in $[0, 1]$ interval.
We achieve this by making an elementwise comparison between the `values` array which is generated by the binomial function and contains how many times each event actually occurs with the input argument `counts`  which contains the maximum number of times each event may occur. 

`
        assert np.greater_equal(np.array(counts), np.array(values)).all()
`

Side note: I couldn't find an equivalent implementation for this test case in keras or numpy. So currently I'm using bare `assert` with numpy's method in the `random_test.py ` file.


**Test 2:** Computes the probability of each event using the generated values and compares it with the corresponding input probability value in the `probabilities` input argument to make sure they approximate the input probabilities.
We achieve this by performing an elementwise division of `values` by `counts` and then use the `self.assertAllClose` with `rtol` and `atol` values of `0.005` because if we want to be more precise we'll have to increase the counts value to millions as higher the number of draws, the more closely will the generated probability approximate the input probability of the event. But for the sake of testing, I believe checking to this precision should be fine as we're already performing tens of thousands of draws. 
`
        generated_probabilities = np.array(values) / np.array(counts)
        probabilities = np.ones(shape) * np.array(probabilities)
        self.assertAllClose(
            probabilities, generated_probabilities, rtol=0.005, atol=0.005
        )
`

The current number of draws we're doing in the unit tests:
Unit test 1: `"counts": 5e4`,
Unit test 2: `"counts: 1e5`,
Unit test 3: `"counts": [[1e5, 2e5, 3e5], [4e5, 5e5, 6e5]]`


I've extensively checked these two tests for all four backends with different parameters and they seem to work well. But please let me know if you spot any issues or room for improvement.